### PR TITLE
Update spinner to use the updated `NewOutputWriterSpinner` API from plugin runtime

### PR DESCRIPTION
**Pull request by anujc25**
### What this PR does / why we need it

- Update spinner to use the updated `NewOutputWriterSpinner ` API from plugin runtime

### Which issue(s) this PR fixes

Fixes #

### Describe testing done for PR
- Tested the `tanzu plugin install` that uses this spinner.

### Release note

```release-note
None
```

### CODE REVIEW

**Comment by anujc25** [comment]
This PR is dependent on https://github.com/vmware-tanzu/tanzu-plugin-runtime/pull/157.

File: go.mod [hunk 1]
```diff
@@ -42,7 +42,7 @@ require (
 	github.com/vmware-tanzu/carvel-ytt v0.40.0
 	github.com/vmware-tanzu/tanzu-cli/test/e2e/framework v0.0.0-00010101000000-000000000000
 	github.com/vmware-tanzu/tanzu-framework/capabilities/client v0.0.0-20230523145612-1c6fbba34686
-	github.com/vmware-tanzu/tanzu-plugin-runtime v1.2.0-dev.0.20240122215628-021d60b78abe
+	github.com/vmware-tanzu/tanzu-plugin-runtime v1.2.0-dev.0.20240204224748-25dae56d7026
```
Comment (anujc25) [inline go.mod:45]: TODO: Needs to be updated once https://github.com/vmware-tanzu/tanzu-plugin-runtime/pull/157 is merged.

File: pkg/pluginmanager/manager.go [hunk 2]
```diff
@@ -767,13 +761,17 @@ func installOrUpgradePlugin(p *discovery.Discovered, version string, installTest
 		}()
 
 		// Initialize the spinner
-		sow, spinnerErr = component.NewOutputWriterSpinnerWithSpinnerOptions(os.Stderr, component.TableOutputType, installingMsg, true, spinnerOptions)
+		spinner, spinnerErr = component.NewSpinner(component.WithStdout(os.Stderr),
+			component.WithOutputFormat(component.TableOutputType),
```
Comment (marckhouzam) [inline pkg/pluginmanager/manager.go:765]: I guess this is something we don't need to set as per your [comment](https://github.com/vmware-tanzu/tanzu-plugin-runtime/pull/157#discussion_r1477468591)

File: pkg/pluginmanager/manager.go [hunk 3]
```diff
@@ -767,13 +761,17 @@ func installOrUpgradePlugin(p *discovery.Discovered, version string, installTest
...
 			log.V(6).Infof("Unable to initialize spinner: %v", spinnerErr.Error())
 			log.Info(installingMsg)
 		}
-		if sow != nil {
-			defer sow.StopSpinner()
+		if spinner != nil {
+			defer spinner.StopSpinner()
 		}
 		// Start a goroutine that listens for interrupt signals
 		go func(s component.OutputWriterSpinner) {
```
Comment (marckhouzam) [inline pkg/pluginmanager/manager.go:777]: Can you change this type to the new `Spinner`?

**Review by marckhouzam — APPROVED** [review]
LGTM with 2 small nits.
Thanks!

File: pkg/pluginmanager/manager.go (hunk 3 above)
Comment (anujc25) [reply pkg/pluginmanager/manager.go:776]: As we have renamed `Spinner` back to `OutputWriterSpinner` in plugin-runtimr this change is not needed anymore.

File: pkg/pluginmanager/manager.go [hunk 4]
```diff
@@ -767,14 +760,13 @@ func installOrUpgradePlugin(p *discovery.Discovered, version string, installTest
...
 		// Initialize the spinner
-		sow, spinnerErr = component.NewOutputWriterSpinnerWithSpinnerOptions(os.Stderr, component.TableOutputType, installingMsg, true, spinnerOptions)
-		if spinnerErr != nil {
-			log.V(6).Infof("Unable to initialize spinner: %v", spinnerErr.Error())
-			log.Info(installingMsg)
-		}
-		if sow != nil {
-			defer sow.StopSpinner()
-		}
+		spinner, _ = component.NewOutputWriterSpinner(component.WithOutputStream(os.Stderr),
```
Comment (marckhouzam) [inline pkg/pluginmanager/manager.go:763]: If the API returns an error then we should handle it. That way if in a future version of the runtime we do return an actual error, this code will be ready.

**Review by marckhouzam — APPROVED** [review]
Once the tests pass LGTM with a small ~adoration~ suggestion

**Comment by marckhouzam** [comment]
> Once the tests pass LGTM with a small ~adoration~ suggestion

Autocorrect changed things on me 😂

File: pkg/pluginmanager/manager.go (hunk 4 above)
Comment (anujc25) [reply pkg/pluginmanager/manager.go]: We updated the API in the PluginRuntime and now it does not return an error. Updated PR accordingly.

**Pull request merged**